python/cuml/dask/datasets/make_regression.py

```diff
@@ -0,0 +1,216 @@
+#
+# Copyright (c) 2019-2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import dask.array as da
+import cupy as cp
+
+
+def make_low_rank_matrix(n_samples=100, n_features=100, effective_rank=10,
+                         tail_strength=0.5, random_state=None, n_parts=1):
+    """ Generate a mostly low rank matrix with bell-shaped singular values
+
+    Parameters
+    ----------
+    n_samples : int, optional (default=100)
+        The number of samples.
+    n_features : int, optional (default=100)
+        The number of features.
+    effective_rank : int, optional (default=10)
+        The approximate number of singular vectors required to explain most of
+        the data by linear combinations.
+    tail_strength : float between 0.0 and 1.0, optional (default=0.5)
+        The relative importance of the fat noisy tail of the singular values
+        profile.
+    random_state : int
+        Determines random number generation for dataset creation. Pass an int
+        for reproducible output across multiple function calls.
+    n_parts : int, optional (default=1)
+        The number of parts of work.
+
+    Returns
+    -------
+    X : Dask-cuPy array of shape [n_samples, n_features]
+        The matrix.
+    """
+    rs = da.random.RandomState(seed=random_state,
```

Comment:
See comment above, I think we should create a cupy random state and use it to initialize the Dask.Array random state (assuming it will then use CuPy to general the Gaussian vectors).